team-telnyx/webrtc
# Generate API docs from TS

Introduces a new package as my proposed solution for syncing TS docs with developer docs. The workflow would be as follows:

1. Auto-generate TS docs as JSON every time `@telnyx/webrtc` is published
2. Make TS doc JSON available via `@telnyx/webrtc-api-docs` package, sync npm version to `@telnyx/webrtc` version
3. Add `@telnyx/webrtc-api-docs` to react-developer-docs
4. Render react-developer-docs based on JSON imported from package

This seems like a sane way to keep versions in sync, instead of storing API versions or even content in another platform like Contentful. You would only need to upgrade the `@telnyx/webrtc-api-docs` dependency in dev docs and rebuild to get the latest version.

Follow up work to consume JSON would be done in [react-developer-docs](https://github.com/team-telnyx/react-developer-docs).

## 📝 To Do


## ✋ Manual testing

1. Run `npm install` in `packages/js` if you haven't done so already
2. Run `npm run docs` in `packages/api-docs`. Verify that JSON output in `packages/api-docs/lib` looks as expected:
```js
{
  "id": 376,
  "name": "VertoMethod",
  "kind": 4,
  "kindString": "Enumeration",
  "flags": {
  "isExported": true
  },
  "children": [
    // ...etc
  ]
}
```
